lua/plenary/path.lua

```diff
@@ -229,6 +240,36 @@ function Path:rmdir()
   uv.fs_rmdir(self:absolute())
 end
 
+function Path:touch(opts)
+  opts = opts or {}
+
+  local mode = opts.mode or 420
+  local parents = F.if_nil(opts.parents, false, opts.parents)
+
+  if self:exists() then return end
+
+  if parents then
+    -- alt: split with self._sep take everything besides last entry. @tj?
+    local only_dirs = self:absolute():match(string.format('^(.+)%s([^%s]+)', self._sep, self._sep))
+    Path:new(only_dirs):mkdir({ parents = true })
```

Comment:
Which version should i implement.
Regex or split?

Reply:
regex is harder to read but I feel always better that split not sure

Reply:
I'd implement `self:parents()` and then just take everything but the last one.

Reply:
(remember, anything that we do in CPU land is infinitely faster than IO :) )